include/transport/time_queue.h

```diff
@@ -401,17 +473,10 @@ namespace qtransport {
                 return _timer_ctx.ticks;
             }
 
-            if (_queue_index && _queue_index >= _queue.size()) {
-                _queue.clear();
-                _queue_index = 0;
-            }
-
-            for (int i = 0; i < _timer_ctx.delta; i++) {
-                _buckets[(_bucket_index + i) % _total_buckets].clear();
-            }
-
+            _buckets[_bucket_index].clear();
```

Comment:
We need to allow those to be there even though time has passed to support the spike. Otherwise, we clear them and allowing spikes doesn't do anything because the buckets are cleared. :)  I'll add back the clearing when latency spike is disabled.   Clearing one appears to still be okay in terms of memory, but we need to watch it. Buckets are still cleared when latency is greater than bucket size and when the queue is being processed with no lag.